From b282e1a88eddc49908367f84b76bf4c45c698fcb Mon Sep 17 00:00:00 2001
From: Anoob Joseph <anoobj@marvell.com>
Date: Tue, 14 Nov 2023 05:51:05 +0000
Subject: [PATCH 053/513] test/security: add sha1-hmac to auth list

Add SHA1-HMAC with 12B digest to the auth list. This algorithm would be
used in IPsec tests and combined tests would run on devices that support
the capabilities.

Signed-off-by: Anoob Joseph <anoobj@marvell.com>
Change-Id: If13e97d5a56d564a5591ff90bcb93e5e5d86fbe3
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/115861
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/117478
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
---
 app/test/test_security_proto.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/app/test/test_security_proto.h b/app/test/test_security_proto.h
index 596402fff9dd3..d6a00018e3c79 100644
--- a/app/test/test_security_proto.h
+++ b/app/test/test_security_proto.h
@@ -99,6 +99,12 @@ static const struct crypto_param auth_list[] = {
 		.key_length = 16,
 		.digest_length = 12,
 	},
+	{
+		.type = RTE_CRYPTO_SYM_XFORM_AUTH,
+		.alg.auth =  RTE_CRYPTO_AUTH_SHA1_HMAC,
+		.key_length = 20,
+		.digest_length = 12,
+	},
 	{
 		.type = RTE_CRYPTO_SYM_XFORM_AUTH,
 		.alg.auth =  RTE_CRYPTO_AUTH_SHA256_HMAC,
-- 
2.25.1

